Authentication/LogIn/View/LoginController.swift

```diff
@@ -48,10 +55,12 @@ public final class LoginController: UIViewController {
     init(viewModel: LoginViewModelType,
         loginViewFactory: () -> LoginViewType,
         onRegister: (LoginController) -> (),
-        delegate: LoginControllerDelegateType = DefaultLoginControllerDelegate()) {
+        onRecoverPassword: (LoginController) -> (),
```

Comment:
I think that this initializer is getting too convoluted. Maybe we can move the `onRecoverPassword` and `onRegister` to its own delegate. The difference being we wont provide a default delegate. This must be provided by the user.